six/modules/c++/six.sicd/source/ComplexXMLParser04x.cpp

```diff
@@ -160,11 +160,23 @@ XMLElem ComplexXMLParser04x::convertImageFormationToXML(
     //! this segment was recreated completely because the ordering of
     //! a lot of the variables has been updated
     XMLElem imageFormationXML = newElement("ImageFormation", parent);
-
+    XMLElem segmentElem = NULL;
```

Comment:
Similarly here, use the C++ structure to check for existence and fix exception message